src/blockdata/constants.rs

```diff
@@ -124,6 +124,21 @@ pub fn genesis_block(network: Network) -> Block {
                 txdata: txdata
             }
         }
+        Network::Signet => {
+            // TODO: somehow obtain the block script and generate coinbase tx from it
+            let txdata = vec![bitcoin_genesis_tx()];
+            Block {
+                header: BlockHeader {
+                    version: 1,
+                    prev_blockhash: Default::default(),
+                    merkle_root: txdata[0].txid(),
+                    time: 1534313275,
+                    bits: 0x1e2adc28,
+                    nonce: 621297
```

Comment:
Note to reviewers: you can verify the `time`, `bits`, and `nonce` parameters by checking: 

https://github.com/kallewoof/bitcoin/blob/2b9db9ac0936cf278ad27c59497d9feec61f6440/src/chainparams.cpp#L512

and

https://github.com/kallewoof/bitcoin/blob/2b9db9ac0936cf278ad27c59497d9feec61f6440/src/chainparams.cpp#L270

We need to make sure that, before merging, there's some way of passing in the blockscript and creating the Coinbase signature like so:

https://github.com/kallewoof/bitcoin/blob/2b9db9ac0936cf278ad27c59497d9feec61f6440/src/chainparams.cpp#L505-L513